interfaces.go

```diff
@@ -21,9 +23,15 @@ type collectorResponse interface {
 	Disable() bool
 }
 
+type reportRequest struct {
+	thriftRequest *lightstep_thrift.ReportRequest
+	grpcRequest   *cpb.ReportRequest
+}
+
 // collectorClient encapsulates internal thrift/grpc transports.
 type collectorClient interface {
-	Report(context.Context, *reportBuffer) (collectorResponse, error)
+	Translate(context.Context, *reportBuffer) (reportRequest, error)
```

Comment:
Thought: Could we separate out this translate code into an `Encoder` interface instead? I created a new `httpCollectorClient` in `v0.15.0` and it would be nice to share the `Encoder` between the two `clients`.

Reply:
@MatthewDolan are you saying just to rename it to Encoder? or are you trying to get at something bigger here?

Reply:
What I think I am trying to say is instead of adding `Translate(...)` to the `collectorClient` interface. Instead add a new interface:

```
type encoder interface {
  Translate(context.Context, *reportBuffer) (reportRequest, error)
}
```

There would be two encoders today (`protoEncoder` and `thriftEncoder`) and there would be three clients (`httpCollectorClient`, `grpcCollectorClient`, and `thriftCollectorClient`). The http and grpc implementations would share the same encoder.

Does that make sense?

Reply:
@MatthewDolan were you thinking of having the converter moved up a level then as well? right now the converter for PROTO lives in the client and in instantiated when the client is. I just want to make sure I'm understanding you correctly. You are proposing that the tracer call encoder.Translate and the encoder is is a part of the tracer and not the client? I know that you made some changes in this area recently so I want to make sure I don't alter something that you've already planned and thought about.

Reply:
We could definitely move it up into the tracer itself. I only put it in the client because I was trying to do the minimum change. I don't have strong feelings about where it lives.